src/App.jsx

```diff
@@ -1,29 +1,52 @@
-import React, { useState } from 'react'
+import React, { useEffect, useState } from 'react'
 import './App.css'
 import './TodoList'
 import TodoList from './TodoList'
 import './AddTodoForm'
 import AddTodoForm from './AddTodoForm'
 
-const useSemiPersistentState = () => {
 
-  const [todoList, setTodoList] = React.useState(() => {
 
-    const savedTodoList = localStorage.getItem('savedTodoList')
-    return JSON.parse(savedTodoList) || [];
-  });
-  
+function App() {
+
+  const [todoList, setTodoList] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(true);
+
+
+
+    React.useEffect(() =>{
```

Comment:
There is no need for the conditional statement in the useEffect function, because the aim of this useeffect is to populate the `todoList` from the local storage at first load. see suggestion below

```suggestion
    React.useEffect(() =>{
      const savedTodoList = JSON.parse(localStorage.getItem('savedTodoList'));
      const fetchData = new Promise((resolve, reject) => {
        setTimeout(() => {
          resolve({data: {todoList: savedTodoList}})
        }, 2000)
      });

      fetchData.then((result) => {
        setTodoList(result.data.todoList);
        setIsLoading(false);
      })
  }, [])

```